### ISSUE TITLE
bedtools Intersect: Received illegal bin number

### ISSUE THREAD

**Issue by olegs** [arq5x/bedtools2#596]
I face this error with bedrolls 2.26.0 installed using `bioconda` on Mac OS.
```
unit-809:rip oleg$ bedtools --version
bedtools v2.26.0
unit-809:rip oleg$ bedtools intersect -a a.bed.txt -b peaks.bed.txt -c -f 0.20
ERROR: Received illegal bin number 4294967295 from getBin call.
ERROR: Unable to add record to tree.
```
See files [a.bed.txt](https://github.com/arq5x/bedtools2/files/1495080/a.bed.txt) and [peaks.bed.txt](https://github.com/arq5x/bedtools2/files/1495081/peaks.bed.txt) in attachment.
**NOTE** `peaks.bed.txt` is not TAB separated file.

Looks like this is a small example for  #525 you were asking for.

**Comment by arq5x** [comment]
@nkindlon - do have a feeling about the best place to improve this error message?  When the file is not tab delimited, the user gets a cryptic error about adding a record to the BIN tree. Instead, it should report that the file is not tab delimited sooner in the process.

**Comment by nkindlon** [comment]
Hi Aaron,
       Looking at the attached peaks.bed.txt file above, I'm not sure how it was recognized by the type checker in the first place. It appears to have a tab in between the chromosome and start position, but then multiple spaces between the start pos and end pos. Is that how you see it? If, so it should fail validation in the type checker before intersect tries to do anything with it.

**Comment by nkindlon** [comment]
Ok, I ran it through the debugger. Because peaks.bed.txt only has one tab, and thus only two distinct fields, it does correctly fail the test for bed format, and is instead recognized as a NO_POS_PLUS_RECORD_TYPE file. I.e, it would work with groupby. In src/utils/fileType/FileRecordTypeChecker.cpp::handleTextFormat, lines 242-244:
		//Here the Record must not have positions, so it is the NoPosPlus Type.
		_recordType = NO_POS_PLUS_RECORD_TYPE;
		return true;


 Refresh my memory, should that format work with any tool except groupby? If not, I can change that return statement to false, since _isGroupBy is already handled earlier in the same function in the last pull request I submittted. Changing it to false would cause the program to exit with the error:
"Error: unable to open file or unable to determine types for file peaks.bed.txt"
